seacrowd/sea_datasets/hse_thai/hse_thai.py

```diff
@@ -0,0 +1,139 @@
+import csv
+import os
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Licenses, Tasks
+
+_CITATION = """\
+    @misc{rtatman2017hse_thai,
+        author = {Rachel Tatman},
+        title = {HSE Thai Corpus},
+        howpublished = {\\url{https://www.kaggle.com/datasets/rtatman/hse-thai-corpus}},
+        note = {Accessed: 2023-11-22}
+    }
+"""
+
+_DATASETNAME = "hse_thai"
+
+_DESCRIPTION = """\
+HSE Thai Corpus is a corpus of modern texts written in Thai language. The texts, containing in whole 50 million tokens, were collected from various Thai websites (mostly news websites). To make it easier for non-Thai-speakers to comprehend and use texts in the corpus the researchers decided to separate words in each sentence with spaces. The data for the corpus was collected by means of Scrapy. To tokenize texts the Pythai module was used. The text in this dataset is encoded in UTF-8. This dataset contains text from two sources: Wikipedia and thaigov.go.th. The former is licensed under a standard Wikipedia license, and the latter under an Open Government License for Thailand.
+
+Before running the dataset, please make sure your CLI can run Kaggle API. Guide for installing: https://github.com/Kaggle/kaggle-api/blob/main/docs/README.md
+"""
+
+_HOMEPAGE = "https://www.kaggle.com/datasets/rtatman/hse-thai-corpus"
+
+_LANGUAGES = ["tha"]
+
+_LICENSE = Licenses.APACHE_2_0.value
+
+_LOCAL = False
+
+_URLS = "rtatman/hse-thai-corpus"
+
+_SUPPORTED_TASKS = [Tasks.LANGUAGE_IDENTIFICATION]
```

Comment:
Another thing. If we looked into the original source description, it seems that the tasks can be extended outside of language identification such as translation and parts of speech tagging:

> This website gives access to the HSE Thai Corpus - the corpus of modern texts written in Thai language. The texts, containing in whole 50 million tokens, were collected from various Thai websites (mostly news websites). Each token was assigned it's English translation and part of speech tag.

Reply:
Yes, but I think it is from the source of the kaggle dataset! But not included in the kaggle dataset.